lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1428,6 +1276,292 @@ def write_store(
+        step_jobs_summary = self.invocations_service.show_invocation_step_jobs_summary(trans, invocation_id)
+        return [InvocationStepJobsResponse(**summary) for summary in step_jobs_summary]
+
+    @router.get(
+        "/api/workflows/{workflow_id}/invocations/{invocation_id}/step_jobs_summary",
+        summary="Get job state summary info aggregated per step of the workflow invocation.",
+    )
+    @router.get(
+        "/api/workflows/{workflow_id}/usage/{invocation_id}/step_jobs_summary",
+        summary="Get job state summary info aggregated per step of the workflow invocation.",
+        deprecated=True,
+    )
+    def workflow_invocation_step_jobs_summary(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+        invocation_id: DecodedDatabaseIdField = InvocationIDPathParam,
+    ) -> List[InvocationStepJobsResponse]:
+        """An alias for `GET /api/invocations/{invocation_id}/step_jobs_summary`. `workflow_id` is ignored."""
+        return self.invocation_step_jobs_summary(trans, invocation_id)
+
+    @router.get(
+        "/api/invocations/{invocation_id}/jobs_summary",
+        summary="Get job state summary info aggregated across all current jobs of the workflow invocation.",
+    )
+    def invocation_jobs_summary(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        invocation_id: DecodedDatabaseIdField = InvocationIDPathParam,
+    ) -> InvocationJobsResponse:
+        """
+        Warning: We allow anyone to fetch job state information about any object they
+        can guess an encoded ID for - it isn't considered protected data. This keeps
+        polling IDs as part of state calculation for large histories and collections as
+        efficient as possible.
+        """
+        jobs_summary = self.invocations_service.show_invocation_jobs_summary(trans, invocation_id)
+        return InvocationJobsResponse(**jobs_summary)
+
+    @router.get(
+        "/api/workflows/{workflow_id}/invocations/{invocation_id}/jobs_summary",
+        summary="Get job state summary info aggregated across all current jobs of the workflow invocation.",
+    )
+    @router.get(
+        "/api/workflows/{workflow_id}/usage/{invocation_id}/jobs_summary",
+        summary="Get job state summary info aggregated across all current jobs of the workflow invocation.",
+        deprecated=True,
+    )
+    def workflow_invocation_jobs_summary(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+        invocation_id: DecodedDatabaseIdField = InvocationIDPathParam,
+    ) -> InvocationJobsResponse:
+        """An alias for `GET /api/invocations/{invocation_id}/jobs_summary`. `workflow_id` is ignored."""
+        return self.invocation_jobs_summary(trans, invocation_id)
+
+    # Should I even create models for those as they will be removed?
```

Comment:
The following 3 endpoints are meant to be removed after 23.1 Should I still create pydantic models for them or not?

Reply:
No, you don't need to, those endpoints will be removed soon™